code/modules/rykka_puzzle_wips/reactor_defense.dm

```diff
@@ -0,0 +1,319 @@
+/*													Object Definition																			*/
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/structure/damaged_reactor // Damaged Reactor object used as the "thing" we interact with
+	name = "damaged reactor"
+	desc = "You shouldn't be able to read this! OOC: Poke @Rykka Stormheart or ask in dev, the reactors should be a subtype of damaged_reactor!"
+	icon = 'icons/obj/damaged_reactor.dmi'
+	icon_state = "reactor_off"
+	density = 1
+	anchored = TRUE
+	climbable = FALSE
+	breakable = FALSE
+	var/icon_state_warmup = null		// Icon to use when the reactor is engaged/warming up
+	var/icon_state_severe_damage = null	// Icon to use when the reactor has taken severe damage.
+	var/icon_state_powered = null		// Icon to use once the reactor is at full power
+	var/active_defense = 0				// Are we actively in "defense" mode? (IE has someone engaged the defense gamemode?)
+	var/health = 2000					// The Reactor's health value
+	var/maxhealth = 2000				// The maximum health this object can have.
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*													End Object Definition																		*/
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+/*	//		=====							Interaction Code Begins															=====		//
+ *	//		=====			Handle Examine, attempting to interact with object, other "generic" code here.					=====		//
+ *////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/obj/structure/damaged_reactor/attack_hand(var/mob/living/user) // Is a LIVING mob clicking on us? Sorry ghosts. :p
+	if(!isliving(user)) // You're not alive. Stop. Bad.
+		return // No clicky.
+	switch(user.a_intent)
+		if(I_HELP)
+			ui_interact(user) // Open NanoUI
+		if(I_HURT)
+			var/mob/living/carbon/human/H = usr
+			if(H.species.can_shred(H)) // This is a hard vore proc. Whyyyyy???
+				attack_generic(H,10) // Handle doing damage here too, since it counts as an interaction.
+				return
+			// else if(H.species.) TODO: Figure out how to handle SHARP claws/etc for a very little bit of damage.
+			else // You can't damage it without a tool!
+				playsound(src, 'sound/effects/grillehit.ogg', 50, 1)
+				user.do_attack_animation(src)
+				user.visible_message("<span class = 'danger'>\The [user] slams their hand against \the [src], not appearing to damage it!</span>",
+									"<span class = 'danger'>You don't appear to do anything to the heavy exterior of \the [src]!</span>")
+		if(I_GRAB)

```

Comment:
Why specifically intercept these intents and print a message? Why not just have any non-harm intent click open the menu?

Reply:
I thought it'd be better to do it this way?

Should I just have the other intents open the menu too?

Reply:
I think it would simplify the interaction and be less of a 'gotcha' but ultimately it's up to you.